Sources/web3swift/Utils/EIP/EIP681.swift

```diff
@@ -281,15 +281,15 @@ extension Web3 {
             } else {
                 code.functionName = components.path
             }
-            guard let queryItems = components.queryItems else {return code}
+            guard let queryItems = components.queryItems else { return code }
             var inputNumber: Int = 0
             var inputs = [ABI.Element.InOut]()
             for comp in queryItems {
                 if let inputType = try? ABITypeParser.parseTypeString(comp.name) {
                     guard let rawValue = comp.value,
                           let functionArgument = await parseFunctionArgument(inputType,
                                                                              rawValue.trimmingCharacters(in: .whitespacesAndNewlines),
-                                                                             chainID: code.chainID ?? 0,
+                                                                             chainID: code.chainID,
```

Comment:
A bug was introduced here by setting default value to 0.
Now, if the `chainID` is `nil` the value of `Networks.Mainnnet.chainID` is used in the `parseFunctionArgument` function.
Note, that the issue is related only to parsing ENS addresses, and to be specifically the host part, e.g. `ethereum:someaddress.eth@8834`. If we do not have the `chainID`, that's the `8834` value in the example we pick 0 as default. Then verification of the `someaddress.eth` fails because of that chain ID value.

I do not think this is the final correct solution.
Instead what must be thought through is the following: what is the default behaviour when chain ID is not provided? Is the Mainnet chain ID a correct default value?

I'll look into this once again to figure out if we need to change this part again.